README.md

```diff
@@ -31,108 +33,136 @@ cur.execute('SELECT * FROM system.runtime.nodes')
 rows = cur.fetchall()
 ```
 
-This will query the `system.runtime.nodes` system tables that shows the nodes
-in the Trino cluster.
+This will query the `system.runtime.nodes` system tables that shows the nodes in the Trino cluster.
 
 The DBAPI implementation in `trino.dbapi` provides methods to retrieve fewer
-rows for example `Cursor.fetchone()` or `Cursor.fetchmany()`. By default
+rows for example `Cursorfetchone()` or `Cursor.fetchmany()`. By default
 `Cursor.fetchmany()` fetches one row. Please set
 `trino.dbapi.Cursor.arraysize` accordingly.
 
-# Basic Authentication
+### 1.3 Authentications
+
+#### Basic Authentication
+
 The `BasicAuthentication` class can be used to connect to a Trino cluster configured with
 the [Password file authentication type, LDAP authentication type or Salesforce authentication type](https://trino.io/docs/current/security/authentication-types.html):
 
 ```python
-import trino
-conn = trino.dbapi.connect(
-    host='coordinator url',
-    port=8443,
-    user='the-user',
-    catalog='the-catalog',
-    schema='the-schema',
+from trino.dbapi import connect
+from trino.auth import BasicAuthentication
+
+conn = connect(
     http_scheme='https',
-    auth=trino.auth.BasicAuthentication("principal id", "password"),
+    auth=BasicAuthentication("principal id", "password"),
```

Comment:
Revert.

The principal is a different thing from the `user` that logs in.

Reply:
Yeah, I know.
Add a note to explain about `username` & `principal_id`

Reply:
Move to `User impersonation` session